From d76a95e29dfed86f98f32a053f88325b14c22d84 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Wed, 8 Apr 2020 15:26:54 +0700
Subject: [PATCH] include: rzg-modules-common: Fix race condition in kernel
 module build

Sometimes a build fails with the message "/bin/sh: scripts/mod/modpost:
Permission denied".

This patch addresses the issue by locking access to kernel_scripts,
meaning only one out-of-tree module can be compiled and modify modpost
at the same time.

The downside is that builds may be slowed down, but it's not expected
that the impage will be large.

The original patch is
"module.bbclass: Fix potential do_compile/do_make_scripts race
condition",
taken from https://marc.info/?l=openembedded-core&m=149475293117017&w=2

More details on
https://marc.info/?l=openembedded-core&m=149423983817820&w=2

Signed-off-by: Chris Paterson <chris.paterson2@renesas.com>
Signed-off-by: Binh Nguyen <binh.nguyen.uw@renesas.com>
Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 meta-rzg1/include/rzg-modules-common.inc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/meta-rzg1/include/rzg-modules-common.inc b/meta-rzg1/include/rzg-modules-common.inc
index 4a3658ae..b21e1fa 100644
--- a/meta-rzg1/include/rzg-modules-common.inc
+++ b/meta-rzg1/include/rzg-modules-common.inc
@@ -7,3 +7,7 @@ export LDFLAGS = ""
 export CP = "cp"
 
 inherit module
+
+# Ensure one recipe isn't running do_make_scripts whilst another is using those
+# scripts in do_compile.
+do_compile[lockfiles] = "${TMPDIR}/kernel-scripts.lock"
-- 
2.7.4

